packages/browser-wallet/src/popup/pages/Account/Earn/Baking/Baking.tsx

```diff
@@ -0,0 +1,199 @@
+import LoadingIcon from '@assets/svg/pending-arrows.svg';
+import { BrowserWalletAccountTransaction, TransactionStatus } from '@popup/shared/utils/transaction-history-types';
+import ButtonGroup from '@popup/shared/ButtonGroup';
+import { openStatusToDisplay } from '@popup/shared/utils/baking-helpers';
+import DisplayPartialString from '@popup/shared/DisplayPartialString';
+import RegisterBaking from './RegisterBaking';
+import RemoveBaking from './RemoveBaking';
+import UpdateBaking from './UpdateBaking';
+import { earnPageContext } from '../utils';
+
+const routes = {
+    register: 'register',
+    update: 'update',
+    remove: 'remove',
+};
+
+type PendingBakingProps = {
+    transaction: BrowserWalletAccountTransaction;
+};
+
+function PendingBaking({ transaction }: PendingBakingProps): JSX.Element {
+    const { t } = useTranslation('account', { keyPrefix: 'baking.details' });
+
+    return (
+        <div className="flex-column align-center justify-center h-full">
+            {transaction.status !== TransactionStatus.Failed && <LoadingIcon className="loading" />}
+            <h3 className="m-t-5 m-b-0 m-h-30 text-center">
+                {t(transaction.status === TransactionStatus.Failed ? 'failed' : 'pending')}
+            </h3>
+        </div>
+    );
+}
+
+type DisplayPendingChangeProps = {
+    pendingChange?: StakePendingChange;
+};
+
+function DisplayPendingChange({ pendingChange }: DisplayPendingChangeProps) {
+    const { t: sharedT } = useTranslation('shared', { keyPrefix: 'baking' });
+    const { consensusStatus, tokenomicsInfo, chainParameters } = useContext(earnPageContext);
+
+    const effectiveTime = useMemo(() => {
+        if (pendingChange) {
+            const date = dateFromStakePendingChange(pendingChange, consensusStatus, tokenomicsInfo, chainParameters);
+            if (date) {
+                return getFormattedDateString(date);
+            }
+        }
+        return undefined;
+    }, [
+        (pendingChange as StakePendingChangeV1)?.effectiveTime?.toString?.(),
+        consensusStatus,
+        tokenomicsInfo,
+        chainParameters,
+    ]);
+
+    if (pendingChange?.change === StakePendingChangeType.ReduceStake) {
+        return (
+            <>
+                <div className="earn-details__header">{sharedT('changesTakesEffectOn', { effectiveTime })}</div>
+                <div className="earn-details__header">{sharedT('pendingChange')}</div>
+                <div className="earn-details__value">{displayAsCcd(pendingChange.newStake)} </div>
+            </>
+        );
+    }
+    if (pendingChange?.change === StakePendingChangeType.RemoveStakeV1) {
+        return (
+            <>
+                <div className="earn-details__header">{sharedT('changesTakesEffectOn', { effectiveTime })}</div>
+                <div className="earn-details__value">{sharedT('pendingRemove')}</div>
+            </>
+        );
+    }
+    return null;
```

Comment:
MIght come down to personal preference, but this would remove some of the redundancy

```suggestion
    if (!pendingChange?.change) {
        return null;
    }
    
    return (
            <>
                <div className="earn-details__header">{sharedT('changesTakesEffectOn', { effectiveTime })}</div>
                {pendingChange?.change === StakePendingChangeType.ReduceStake && (
                    <>
                    <div className="earn-details__header">{sharedT('pendingChange')}</div>
                    <div className="earn-details__value">{displayAsCcd(pendingChange.newStake)} </div>
                    </>
                )}
                {pendingChange?.change === StakePendingChangeType.RemoveStakeV1 && (
                    <>
                    <div className="earn-details__header">{sharedT('pendingRemove')}</div>
                    </>
                )}
            </>
        );
```